src/get-started/java-binding.md

```diff
@@ -491,75 +493,201 @@ as specified in [How to Run a Service section](#how-to-run-a-service).
+
+TestKit provides [`getTransactionPool()`][testkit-get-pool] and
+[`findTransactionsInPool(Predicate<TransactionMessage> predicate)`][testkit-find-in-pool]
+methods to inspect transaction pool. It is useful when there is a need to verify
+transactions submitted in `afterCommit` method, as those are put into the pool.
+
+!!! note
+    Note that blocks that are created with
+    [`TestKit.createBlockWithTransactions(Iterable<TransactionMessage> transactionMessages)`][testkit-create-block]
+    will ignore in-pool transactions. As of 0.7.0, there is no way to create a block
+    that would contain both given and in-pool transactions with a single
+    method - to do that, put given transactions into TestKit transaction pool
+    with [`Node.submitTransaction(RawTransaction rawTransaction)`][node-submit-transaction].
+
+#### Checking the blockchain state
+
+In order to verify changes in blockchain state TestKit provides a snapshot of
+the current database state (i.e., the one that corresponds to the latest
+committed block). There are several ways to access it:
+
+- `void withSnapshot(Consumer<Snapshot> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state
+- `<ResultT> ResultT applySnapshot(Function<Snapshot, ResultT> snapshotFunction)`
+  Performs a given function with a snapshot of the current database state and
+  returns a result of its execution
+- `Snapshot getSnapshot()`
+  Returns a snapshot of the current database state
+
+!!! note
+    Note that `withSnapshot` and `applySnapshot` methods destroy the snapshot
+    once the passed closure completes, compared to `getSnapshot`, which
+    disposes created snapshots only when TestKit is closed.
+    Therefore it is recommended to use first two methods if a large number
+    (e.g. more than a hundred) of snapshots needs to be created.
+
+### Time Oracle Testing
+
+The TestKit allows to use [Time Oracle](../advanced/time.md) in your tests.
+To do that, TestKit should be created with
+[`TimeProvider`][testkit-time-provider], which allows to manually manipulate
+time that is returned by TestKit time service.
+
+```java
+ZonedDateTime initialTime = ZonedDateTime.of(2000, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
+TimeProvider timeProvider = FakeTimeProvider.create(initialTime);
+try (TestKit testKit = TestKit.builder()
+      .withService(MyServiceModule.class)
+      .withTimeService(timeProvider)
+      .build()) {
+  Snapshot view = testKit.getSnapshot();
+  TimeSchema timeSchema = TimeSchema.newInstance(view);
+
+  // Commit two blocks for time oracle to update consolidated time. Two blocks
+  // are needed as after the first block time transactions are generated and
+  // after the second one they are processed
+  testKit.createBlock();
+  testKit.createBlock();
+
+  // Check that time service returns time provided by user
+  Optional<ZonedDateTime> consolidatedTime = timeSchema.getTime().toOptional();
+  assertThat(consolidatedTime).hasValue(initialTime);
```

Comment:
Given the users need to know how time service work to use it successfully, shall we expand the example to show this delay in time? Sth like that (with some methods extracted, better variable names, more detailed comments):

```java
  ZonedDateTime initialTime = ZonedDateTime.of(2000, 1, 1, 1, 1, 1, 1, ZoneOffset.UTC);
  // …
  testKit.createBlock(); 
  // The time service have submitted its first transaction, but it has not executed yet
  Optional<ZonedDateTime> consolidateTime1 = testKit.applySnapshot(s -> {
    TimeSchema timeSchema = TimeSchema.newInstance(s);
    return timeSchema.getTime().toOptional();
  });
  // No time available till the time service transaction is processed
  assertThat(consolidatedTime).isEmpty();
  
  // Advance the time
  ZonedDateTime time1 = initialTime.plusSeconds(1);
  timeProvider.setTime(time1);
  testKit.createBlock();
  // The time service have submitted its second transaction. The first must have
  // been executed, with consolidated time now available and equal to initialTime:
  Optional<ZonedDateTime> consolidateTime2 = testKit.applySnapshot(s -> {
      TimeSchema timeSchema = TimeSchema.newInstance(s);
      return timeSchema.getTime().toOptional();
  });
  assertThat(consolidatedTime).hasValue(initialTime);
  
  // Advance the time
  ZonedDateTime time2 = initialTime.plusSeconds(1);
  timeProvider.setTime(time2);
  testKit.createBlock();
  // The time service have submitted its third transaction, and processed the second. 
  // The consolidated time must be equal to time1.
  Optional<ZonedDateTime> consolidateTime3 = testKit.applySnapshot(s -> {
      TimeSchema timeSchema = TimeSchema.newInstance(s);
      return timeSchema.getTime().toOptional();
  });
  assertThat(consolidatedTime).hasValue(time1);
```